config/application.rb

```diff
@@ -151,6 +151,14 @@ class Application < Rails::Application
       Vmdb::Loggers.apply_config(::Settings.log)
     end
 
+    # register plugins even before loading settings, as plugins can bring their own settings
+    initializer :register_vmdb_plugins, :before => :load_vmdb_settings do
+      Rails.application.railties.each do |railtie|
+        next unless railtie.class.name.start_with?("ManageIQ::Providers::") || railtie.try(:vmdb_plugin?)
+        Vmdb::Plugins.instance.register_vmdb_plugin(railtie)
+      end
+    end
+
```

Comment:
Can you put it literally before the other initializer as well within this file?  I'm trying to keep this file in the relative order that it runs so it's less confusing overall